src/jaxsim/rbda/contacts/relaxed_rigid.py

```diff
@@ -390,124 +425,153 @@ def continuing_criterion(carry):
 
             Args:
-                penetration: penetration in constraint frame
-                velocity: velocity in constraint frame
+                pos: position in constraint frame.
+                vel: velocity in constraint frame.
 
             Returns:
+                ξ: computed impedance
                 a_ref: offset acceleration in constraint frame
-                R: regularization matrix
                 K: computed stiffness
                 D: computed damping
             """
-            position = jnp.zeros(shape=(3,)).at[2].set(penetration)
 
-            imp_x = jnp.abs(position) / width
-            imp_a = (1.0 / jnp.power(mid, p - 1)) * jnp.power(imp_x, p)
+            imp_x = jnp.abs(pos) / width
 
+            imp_a = (1.0 / jnp.power(mid, p - 1)) * jnp.power(imp_x, p)
             imp_b = 1 - (1.0 / jnp.power(1 - mid, p - 1)) * jnp.power(1 - imp_x, p)
-
             imp_y = jnp.where(imp_x < mid, imp_a, imp_b)
 
-            imp = jnp.clip(ξ_min + imp_y * (ξ_max - ξ_min), ξ_min, ξ_max)
-            imp = jnp.atleast_1d(jnp.where(imp_x > 1.0, ξ_max, imp))
+            # Compute the impedance.
+            ξ = ξ_min + imp_y * (ξ_max - ξ_min)
+            ξ = jnp.clip(ξ, ξ_min, ξ_max)
+            ξ = jnp.where(imp_x > 1.0, ξ_max, ξ)
 
-            # When passing negative values, K and D represent a spring and damper, respectively.
-            K_f = jnp.where(K < 0, -K / ξ_max**2, 1 / (ξ_max * Ω * ζ) ** 2)
-            D_f = jnp.where(D < 0, -D / ξ_max, 2 / (ξ_max * Ω))
+            # Compute the spring and damper parameters during runtime from the
+            # impedance and other contact parameters.
+            K = 1 / (ξ_max * Ω * ζ) ** 2
+            D = 2 / (ξ_max * Ω)
 
-            a_ref = -jnp.atleast_1d(D_f * velocity + K_f * imp * position)
+            # If the user specifies K and D and they are negative, the computed `a_ref`
+            # becomes something more similar to a classic Baumgarte regularization.
+            K = jnp.where(K < 0, -K / ξ_max**2, K)
+            D = jnp.where(D < 0, -D / ξ_max, D)
 
-            return imp, a_ref, jnp.atleast_1d(K_f), jnp.atleast_1d(D_f)
+            # Compute the reference acceleration.
+            a_ref = -(D * vel + K * ξ * pos)
+
+            return ξ, a_ref, K, D
 
-        def _compute_row(
+        def compute_row(
             *,
-            link_idx: jtp.Float,
-            penetration: jtp.Array,
-            velocity: jtp.Array,
-        ) -> tuple[jtp.Array, jtp.Array]:
+            link_idx: jtp.Int,
+            pos: jtp.Vector,
+            vel: jtp.Vector,
+        ) -> tuple[jtp.Vector, jtp.Matrix, jtp.Float, jtp.Float]:
 
             # Compute the reference acceleration.
-            ξ, a_ref, K, D = _imp_aref(
-                penetration=penetration,
-                velocity=velocity,
+            ξ, a_ref, K, D = imp_aref(
+                pos=pos,
+                vel=vel,
             )
```

Comment:
Probably by removing a comma we can put this in a single line
```suggestion
            ξ, a_ref, K, D = imp_aref(pos=pos, vel=vel)
```

Reply:
Yep, done in https://github.com/ami-iit/jaxsim/pull/260/commits/85a98dabe7afa88ef9b7c913d4b8875e2346e244.